File: 2026-03/src/day_11/boj_11872/Main.java
package day_11.boj_11872;

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int nimsum = 0;

        int n = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        while (n-- > 0) {
            int p = Integer.parseInt(st.nextToken());

            if (p % 4 == 0) {
                nimsum ^= p - 1;
            } else if (p % 4 == 3) {
                nimsum ^= p + 1;
            } else {
                nimsum ^= p;
            }
        }

        if (nimsum != 0) {
            System.out.println("koosaga");
        } else {
            System.out.println("cubelover");
        }
    }
}

File: 2026-03/src/day_11/boj_11872/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int nimsum = 0;

    int n;
    cin >> n;

    while (n--) {
        int p;
        cin >> p;

        if (p % 4 == 0) {
            nimsum ^= p - 1;
        } else if (p % 4 == 3) {
            nimsum ^= p + 1;
        } else {
            nimsum ^= p;
        }
    }

    if (nimsum) {
        cout << "koosaga";
    } else {
        cout << "cubelover";
    }
}
